src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,803 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import javafx.application.Application;
+
+/**
+ * Contains utility functions to be used in several contexts.
+ */
+
+class Utils {
+    /**
+     * Takes an array of strings, a start and end index. Returns
+     * a string with all tokens between start and end (not inclusive)
+     * index separated by a space.
+     *
+     * @param a Array of strings
+     * @param start  start index
+     * @param end end index
+     * @returns String with all tokens between start and end (not inclusive) index separated by space
+     */
+    public static String splitStringBySpaces(String[] a, int start, int end) {
+        String res = "";
+        for (int i = start; i < end; i++) {
+            res += a[i];
+            if (i != end - 1) {
+                res += " ";
+            }
+        }
+        return res;
+    }
+
+    /**
+     * Returns the number of words in a string, separated by spaces.
+     *
+     * @param s String whose words are to be counted
+     * @returns Number of words in the string
+     */
+
+    public static int getWordCount(String s) {
+        return s.split("\\s+").length;
+    }
+}
+
+
+
+/**
+ * Represents a type of task to be done.
+ */
+
+class TaskType {
+    private String task;
+    private boolean isCompleted;
+    private String dateString;
+
+    /**
+     * Constructor for the TaskType class.
+     *
+     * @param task Description of task
+     * @param isCompleted Whether of not the task has been completed
+     * @param dateString String containing only datetime-related information pf the task, if any.
+     */
+
+    public TaskType(String task, boolean isCompleted, String dateString) {
+        this.task = task;
+        this.isCompleted = isCompleted;
+        this.dateString = dateString;
+    }
+
+    /**
+     * Returns the boolean value of the isCompleted field.
+     */
+
+    public boolean getIsCompleted() {
```

Comment:
The method name should also sound like boolean according to the coding standards. Perhaps use isCompleted().